packages/client/src/hooks/useAutoBattle.ts

```diff
@@ -1,52 +1,36 @@
 import dayjs from "dayjs";
-import { useEffect, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 import { useAutoBattleFn } from "./useAutoBattleFn";
 
 export function useAutoBattle() {
   const { autoBattleFn } = useAutoBattleFn();
   const [shouldRun, setShouldRun] = useState<boolean>(false);
-  const [isRunning, setIsRunning] = useState<boolean>(false);
-  const [runningStart, setRunningStart] = useState<number>(0);
+  const tick = useRef(shouldRun);
 
-  useEffect(() => {
-    if (shouldRun) {
-      if (!isRunning) {
-        // set to now
-        setRunningStart(dayjs().unix());
-        setIsRunning(true);
+  tick.current = shouldRun;
 
-        // delay 0.5s to avoid on-chain fail and minimal tick interval
-        setTimeout(() => {
-          // run auto battle
-          autoBattleFn()
-            .then(() => {
-              setIsRunning(false);
-            })
-            .catch((e) => {
-              setIsRunning(false);
-              console.error(e);
-            });
-        }, 500);
-      }
+  const runAutoBattle = () => {
+    if (tick.current) {
+      setTimeout(() => {
+        autoBattleFn()
+          .then(() => {
+            console.log("auto run");
+            runAutoBattle();
+          })
+          .catch((e) => {
+            console.error(e);
+          });
+      }, 500);
     }
-  }, [shouldRun, isRunning, setIsRunning, autoBattleFn, setRunningStart]);
+  };
```

Comment:
What's the problem with old autobattle function

Reply:
This will cause redundant renders and send out multiple requests instantly

Reply:
I reserve my opinion for now, and hope the new code you wrote can solve the problem